framework/transport/ieee1905_transport/test/ieee1905_transport_broker_tests.cpp

```diff
@@ -0,0 +1,320 @@
+{
+    // Configure easylogging++ formatting
+    el::Configurations defaultConf;
+    defaultConf.setToDefault();
+    defaultConf.setGlobally(el::ConfigurationType::Format,
+                            "%level %datetime{%H:%m:%s:%g} <%thread> %fbase[%line] --> %msg");
+
+    // el::Loggers::addFlag(el::LoggingFlag::ColoredTerminalOutput);
+    el::Loggers::reconfigureLogger("default", defaultConf);
+}
+
+TEST(broker_server, invalid_message_magic)
+{
+    BrokerServerWrapper broker_wrapper(broker_uds_file);
+
+    // Register a dummy internal message handler
+    broker_wrapper.register_internal_message_handler(
+        [](std::unique_ptr<mapf::Message> &msg, BrokerServer &broker) -> bool { return true; });
+
+    // Create a random message
+    mapf::Message dummy;
+
+    // Invalid header
+    mapf::Message::Header header;
+    header.magic = INVALID_MAGIC;
+
+    SocketClient sock1(broker_uds_file);
+    ASSERT_EQ(1, broker_wrapper.run()); // Accept the connection
+
+    LOG(DEBUG) << "Sending INVALID magic...";
+    ASSERT_TRUE(messages::send_transport_message(sock1, dummy, &header));
+    ASSERT_EQ(1, broker_wrapper.run()); // Process
+    ASSERT_TRUE(broker_wrapper.error());
+
+    LOG(DEBUG) << "Sending VALID magic...";
+    ASSERT_TRUE(messages::send_transport_message(sock1, dummy));
+    ASSERT_EQ(1, broker_wrapper.run());
+    ASSERT_FALSE(broker_wrapper.error());
+}
+
+TEST(broker_server, subscribe_empty_message)
+{
+    BrokerServerWrapper broker_wrapper(broker_uds_file);
+
+    // Create a subscribe message
+    SubscribeMessage subscribe;
+    subscribe.metadata()->type = SubscribeMessage::ReqType::SUBSCRIBE;
+
+    // Connect to the broker and send the message
+    SocketClient sock1(broker_uds_file);
+    ASSERT_EQ(1, broker_wrapper.run()); // Accept the connection
+
+    ASSERT_TRUE(messages::send_transport_message(sock1, subscribe));
+    ASSERT_EQ(1, broker_wrapper.run()); // Process
+
+    ASSERT_TRUE(broker_wrapper.error());
+}
+
+TEST(broker_server, subscribe_single_type)
+{
+    BrokerServerWrapper broker_wrapper(broker_uds_file);
+
+    // Create a subscribe message
+    SubscribeMessage subscribe;
+    subscribe.metadata()->type               = SubscribeMessage::ReqType::SUBSCRIBE;
+    subscribe.metadata()->msg_types_num      = 1;
+    subscribe.metadata()->msg_types[0].value = 5;
+
+    // Connect to the broker and send the message
+    SocketClient sock1(broker_uds_file);
+    ASSERT_EQ(1, broker_wrapper.run()); // Accept the connection
+
+    ASSERT_TRUE(messages::send_transport_message(sock1, subscribe));
+    ASSERT_EQ(1, broker_wrapper.run()); // Process
+
+    ASSERT_FALSE(broker_wrapper.error());
+}
+
+TEST(broker_server, subscribe_multiple_types)
+{
+    BrokerServerWrapper broker_wrapper(broker_uds_file);
+
+    // Create a subscribe message
+    SubscribeMessage subscribe;
+    subscribe.metadata()->type               = SubscribeMessage::ReqType::SUBSCRIBE;
+    subscribe.metadata()->msg_types_num      = 3;
```

Comment:
```suggestion
    subscribe.metadata()->msg_types_count      = 3;
```

the more I see it in the code the more I find this to be better naming. (was able to strain myself only this far :-) )
(nitpick)

Reply:
Ok, changed